code/modules/mob/living/simple_animal/friendly/ratking.dm

```diff
@@ -0,0 +1,296 @@
+#define RAT_MAYOR_LEVEL 1
+#define RAT_BARON_LEVEL 3
+#define RAT_DUKE_LEVEL 5
+#define RAT_KING_LEVEL 10
+#define RAT_EMPEROR_LEVEL 20
+#define RAT_SAVIOR_LEVEL 30
+#define RAT_GOD_LEVEL 50
+
+
+/proc/announceToRodents(var/message)
+	for(var/R in world_mouses)
+		R << message
+
+// Since there's no proc for the current object to check crossed objects
+/mob/living/simple_animal/mouse/Crossed(atom/movable/O)
+	..()
+
+	if( !health )
+		return
+
+	if( istype(O,/mob/living/simple_animal/mouse/king))
+		var/mob/living/simple_animal/mouse/king/K = O
+		if( !K.health )
+			return
+
+		src.visible_message("<span class='warning'>[src] joins the [K.swarm_name] of \the [K]</span>", \
+							"<span class='notice'>We join our brethren in \the [K.swarm_name]. Long live \the [K].<span>")
+		K.absorb(src)
+
+/mob/living/simple_animal/mouse/king
+	attacktext = "bitten"
+	a_intent = "harm"
+
+	icon_state = "mouse_gray"
+	item_state = "mouse_gray"
+	icon_living = "mouse_gray"
+	icon_dead = "mouse_gray_dead"
+	icon_rest = "mouse_gray_sleep"
+
+	var/swarm_name = "peasentry"
+	var/announce_name = "Request"
+	var/list/rats = list()
+
+/mob/living/simple_animal/mouse/king/New()
+	..()
+
+	update()
+
+	icon_state = initial(icon_state)
+	icon_living = initial(icon_living)
+	icon_dead = initial(icon_dead)
+	body_color = "gray"
+
+	say_dead_direct("An heir to the rat throne has risen, all rejoice and celebrate.")
+	announceToRodents("<span class='notice'>The rat king has risen! Go at once and join his kingdom, long live the king!</span>")
+
+/mob/living/simple_animal/mouse/king/death()
+	while(rats.len)
+		eject(rats[1], 1)
+
+	..()
+
+/mob/living/simple_animal/mouse/king/Move()
+	..()
+
+	for(var/mob/living/simple_animal/mouse/rat in rats)
+		rat.dir = src.dir

```

Comment:
Why is this necessary? The rats are located inside the rat king mob, they shouldn't be visible.